src/communication/websocket/session.cpp

```diff
@@ -46,14 +50,30 @@ void Session::Write(std::shared_ptr<std::string> message) {
   }
   boost::asio::dispatch(strand_, [message = std::move(message), shared_this = shared_from_this()]() mutable {
     shared_this->messages_.push_back(std::move(message));
-
+    if (!shared_this->authenticated_) {
+      return;
+    }
```

Comment:
The check should go before the push_back, `DoWrite` is just the operation to write message from the queue. You still pushed the message to the queue and it will be written to websocket, just not immediatelly. That is not something we want, we want to forbid pushing the message to the queue if the user is not authorized.

Also, I would move the `connected_` check also inside this lambda, just to have a better thread-safety.

Reply:
Whoops, you are right... Will move the connected check as well